platform/lib/middleware/redirects.js

```diff
@@ -28,6 +29,33 @@ const WWW_PREFIX = 'www.';
 const REDIRECT_LINKS_DEFINITION = join(__dirname, '../../config/amp-dev-redirects.yaml');
 const redirectLinks = yaml.safeLoad(readFileSync(REDIRECT_LINKS_DEFINITION));
 
+const AVAILABLE_LOCALES = config.getAvailableLocales();
+const LANGUAGE_PATH_PATTERN = /^\/([a-z]{2}(_[a-z]{2})?)\//;
+
+/**
+ *
+ * @param req
+ * @returns The redirect link or undefined if none was found.
+ */
+function getRedirectLink(req) {
```

Comment:
It'd be good to have a test for this

Reply:
So true.
I hesitated yesterday, because I wanted to have specific unit test rules and it was not obvious how to do that.
Will not happen again :-)